activerecord/test/cases/associations/inverse_associations_test.rb

```diff
@@ -85,6 +87,17 @@ def test_has_many_and_belongs_to_should_find_inverse_automatically_for_sti
     assert_equal post_reflection, author_child_reflection.inverse_of, "The Author reflection's inverse should be the Post reflection"
   end
 
+  def test_has_one_and_belongs_to_with_non_default_foreign_key_should_not_find_inverse_automatically
+    user = User.create!
+    owned_room = Room.create!(owner: user)
+
+    assert_nil user.room
+    assert_nil owned_room.user
+
+    assert_equal user, owned_room.owner
+    assert_equal owned_room, user.owned_room
+  end
+
```

Comment:
Can you also test `reflection.inverse_of`?

```ruby
  def test_has_one_and_belongs_to_with_custom_association_name_should_not_find_wrong_inverse_automatically
    user_reflection = Room.reflect_on_association(:user)
    owner_reflection = Room.reflect_on_association(:owner)
    room_reflection = User.reflect_on_association(:room)

    assert_predicate user_reflection, :has_inverse?
    assert_equal room_reflection, user_reflection.inverse_of

    assert_not_predicate owner_reflection, :has_inverse?
    assert_not_equal room_reflection, owner_reflection.inverse_of
  end
```

Reply:
Thanks, I added.